modules/core/src/ocl.cpp

```diff
@@ -1846,7 +1847,7 @@ struct Queue::Impl
         if( !dh )
             dh = (cl_device_id)pc->device(0).ptr();
         cl_int retval = 0;
-        handle = clCreateCommandQueue(ch, dh, 0, &retval);
+        handle = clCreateCommandQueue(ch, dh, CL_QUEUE_PROFILING_ENABLE, &retval);
```

Comment:
yes, this flag is optional, in ocl4dnn we need this flag to get cl profiling info, but it looks that we don't have a way to set this flag through any interface.  Before we use getenv("ENABLE_OPENCL_PROFILING") to control this flag,  but it doesn't look like a good way because buildbot  doesn't recognize this flag.

Reply:
we add one build option for enabling OpenCL profiling event, it gives user option to control this flag.